packages/mongodb-rag-core/src/dataSources/normalizeUrl.ts

```diff
@@ -0,0 +1,49 @@
+type NormalizeUrlParams = {
+  url: string;
+  removeHash?: boolean;
+  removeQueryString?: boolean;
+};
+
+// Regex used to get just the "front part" of a URL
+const optionalRegex = {
+  REMOVE_HASH: /^[^#]+/,
+  REMOVE_QUERY: /^[^?]+/,
+  REMOVE_BOTH: /^[^?#]+/,
+};
+
+/**
+  Utility function that normalizes a URL.
+  Removes http/s protocol, www, trailing backslashes.
+  Optionally removes query string and hash fragment.
+*/
+export function normalizeUrl({
```

Comment:
We do some URL comparisons in `mongodb-chatbot-ui` - specifically thinking of `messageLinks.ts` and `references.ts`. Can you check if using this throughout the server will affect those comparisons and adjust as needed?